Improve the section about re-starting the miner (#418)
filecoin-project/filecoin-docs: docs/mine/lotus/miner-lifecycle.md

## Safely restarting the miner daemon

The process of shutting down a miner and starting it again is complex as there are several factors that need to be taken into account to be able to do it with all the guarantees:

- How long the miner plans to be offline
- The existence and distribution of proving deadlines for the miner
- The presence of open payment channels and ongoing retrieval deals
- The occurrence of ongoing sealing operations

### Reducing the time offline

Usually, the miner should be offline as little as possible, given the need to continuously send proofs to the network. This includes the time it takes for the miner to fully boot up once it is restarted. For these reasons, it is recommended to:

1. Rebuild, install any upgrades before restarting the Lotus Miner process.
2. Make sure that the proof parameters that are downloaded the first time the miner is started to `/var/tmp/filecoin-proof-parameters` (or to `$FIL_PROOFS_PARAMETER_CACHE` when defined) are in a fast storage drive (NVMe), as they need to be read and verified every time the miner starts.

### Ensuring proofs for the current deadline have been sent

> Before shutting down the miner daemon, verify that no pending operations that could get your miner _[slashed](../slashing.md)_.
> 
> Run `lotus-miner proving info`: If any deadline shows a _block height_ in the past, do not restart.

Reviewer: ```suggestion
Shutting down your miner while there are still pending operations could get your [miner slashed](../slashing.md). Check that there are no pending operations by running `lotus-miner proving info`. If any deadline shows a _block height_ in the past, do not restart yet.
```

Deadline FaultCutoff: 588 (in 45m30s)
```

The `proving info` examples above show information for the current proving window and deadlines. If you wish to see any upcoming deadlines you can use:

```bash
$ lotus-miner proving deadlines
```

Every row corresponds to a deadline (a period of 30 minutes covering 24 hours). The current one is marked. This is sometimes useful to find a time of day in which the miner does not have to submit any proofs to the chain.

### Checking and temporally disabling deals

> Before stopping the miner, you may check the state of your deals to make sure the miner is not receiving data or retrieving data for a client:

Reviewer: ```suggestion
Before stopping the miner, check the state of your deals to make sure the miner is not receiving data or retrieving data for a client:
```

```bash
lotus-miner storage-deals list
lotus-miner retrieval-deals list
lotus-miner data-transfers list
```

> To prevent new deals from coming in while you wait to finish work for the current deadline etc, you may want to disable storage and retrieval deals so that the miner does not find itself in the middle of a new one when shutdown:

Reviewer: ```suggestion
To prevent new deals from coming in while you wait to finish work for the current deadline, you can disable storage and retrieval deals. This ensures that the miner does not find itself in the middle of a new deal when shutting down:
```

```bash
lotus-miner storage-deals selection reject --online --offline
lotus-miner retrieval-deals selection reject --online --offline
```

> After re-start, they can be reset with:

Reviewer: ```suggestion
After the miner has finished rebooting, the deals can be reset with:
```

```bash
lotus-miner storage-deals selection reset
lotus-miner retrieval-deals selection reset
```

### Checking ongoing seal operations

> You should get an overview of your current sectors and states with:

Reviewer: ```suggestion
To get an overview of your current sectors and states, run:
```

```bash
lotus-miner sectors list
```

> Any ongoing sealing operation will be restarted from the last checkpoint (usually corresponds to the start of the current sealing phase). Given that sealing is time consuming, you may decide to wait for some soon-to-finish stages to happen before the restart.

Reviewer: ```suggestion
Any ongoing sealing operation will be restarted from the last checkpoint, and usually corresponds to the start of the current sealing phase. Given that sealing is time consuming, you should wait for some stages that are close to finishing before restarting your miner.
```
